homeassistant/components/opentherm_gw/binary_sensor.py

```diff
@@ -292,36 +214,60 @@ async def async_setup_entry(
     """Set up the OpenTherm Gateway binary sensors."""
     gw_hub = hass.data[DATA_OPENTHERM_GW][DATA_GATEWAYS][config_entry.data[CONF_ID]]
 
-    async_add_entities(
-        OpenThermBinarySensor(gw_hub, source, description)
-        for sources, description in BINARY_SENSOR_INFO
-        for source in sources
-    )
+    for klass, descriptions in (
+        (OpenThermBoilerBinarySensorEntity, BOILER_BINARY_SENSOR_DESCRIPTIONS),
+        (OpenThermGatewayBinarySensorEntity, GATEWAY_BINARY_SENSOR_DESCRIPTIONS),
+        (OpenThermThermostatBinarySensorEntity, THERMOSTAT_BINARY_SENSOR_DESCRIPTIONS),
+    ):
+        async_add_entities(klass(gw_hub, description) for description in descriptions)
 
 
-class OpenThermBinarySensor(OpenThermEntity, BinarySensorEntity):
+class OpenThermBinarySensor(OpenThermBaseEntity, BinarySensorEntity):
     """Represent an OpenTherm Gateway binary sensor."""
 
     entity_description: OpenThermBinarySensorEntityDescription
 
     def __init__(
         self,
         gw_hub: OpenThermGatewayHub,
-        source: str,
         description: OpenThermBinarySensorEntityDescription,
     ) -> None:
         """Initialize the binary sensor."""
+        super().__init__(gw_hub, description)
         self.entity_id = async_generate_entity_id(
```

Comment:
I think this will be breaking though, as I chose quite a harsh approach on the migration (delete everything and re-add it afterwards)... Can we do this after the migration has been released for a few months or should I rewrite the migration to be more specific?